plasmapy/particles/_factory.py

```diff
@@ -44,10 +44,10 @@ def _generate_particle_factory_error_message(
 def _make_custom_particle_with_real_charge_number(
     arg: ParticleLike,
     *,
-    mass_numb: Optional[Integral] = None,
+    mass_numb: Optional[int] = None,
```

Comment:
I was surprised that objects from `numbers` aren't intended to be used with static type checking with mypy ([PEP 484](https://peps.python.org/pep-0484/#the-numeric-tower)).  There are constructs like `typing.SupportsInt` which can be used. 

```
plasmapy/particles/tests/test_atomic.py:417: note: Types from "numbers" aren't supported for static type checking
plasmapy/particles/tests/test_atomic.py:417: note: See https://peps.python.org/pep-0484/#the-numeric-tower
plasmapy/particles/tests/test_atomic.py:417: note: Consider using a protocol instead, such as typing.SupportsFloat
```